iroh-bytes/src/store/file.rs

```diff
@@ -1469,16 +1484,19 @@ impl Actor {
 
     fn run_batched(mut self) -> ActorResult<()> {
         let mut msgs = PeekableFlumeReceiver::new(self.state.msgs.clone());
-        while let Some(msg) = msgs.peek() {
-            if let ActorMessage::Shutdown = msg {
+        while let Some(msg) = msgs.recv() {
+            if let ActorMessage::Shutdown { tx } = msg {
+                if let Some(tx) = tx {
+                    tx.send(()).ok();
+                }
                 break;
             }
             match msg.category() {
                 MessageCategory::TopLevel => {
-                    let msg = msgs.recv().expect("just peeked");
                     self.state.handle_toplevel(&self.db, msg)?;
                 }
                 MessageCategory::ReadOnly => {
+                    msgs.push_back(msg).expect("just recv'd");
```

Comment:
how does this `push_back` construction work?

Reply:
It's just a single option in front of the flume channel to make it peekable. You can push back if you just pulled. Makes structuring some code nicer. If the option is already full this will fail, but we just pulled from it so we know it is not full...